src/GRA.Controllers/AvatarController.cs

```diff
@@ -59,12 +59,70 @@ public async Task<IActionResult> Index()
                 .GroupBy(_ => _.GroupId)
                 .Select(_ => _.ToList())
                 .ToList();
+            var usersresult = _avatarService.GetUserUnlockBundles().Result;
```

Comment:
It appears this method is `async`, so it must be called with `await`, e.g. `await _avatarService.GetUserUnlockBundles()`. It also should be renamed to `GetUserUnlockBundlesAsync()` so that it's clear that it's an async method.

You can also simplify the assignment of the tuple like this:
```
var (userBundles, hasBeenViewed) = await _avatarService.GetUserUnlockBundles();
```
Note the camel case on the variable names.

Reply:
This still needs to be `var usersResult = await _avatarService.GetUserUnlockBundles();` - the dictionary looks great.